perf_test/batched/KokkosBatched_Test_BlockTridiagDirect.cpp

```diff
@@ -75,40 +67,161 @@ typedef Vector<SIMD<value_type>, internal_vector_length> internal_vector_type;
 #endif
 
-template <typename ActiveMemorySpace>
+template <typename ExecutionSpace>
 struct FactorizeModeAndAlgo;
 
-template <>
-struct FactorizeModeAndAlgo<Kokkos::HostSpace> {
+struct FactorizeModeAndAlgoHostImpl {
   typedef Mode::Serial mode_type;
   typedef Algo::Level3::Blocked algo_type;
 };
 
-#if defined(KOKKOS_ENABLE_CUDA)
+#if defined(KOKKOS_ENABLE_SERIAL)
+template <>
+struct FactorizeModeAndAlgo<Kokkos::Serial> : FactorizeModeAndAlgoHostImpl {};
+#endif
+
+#if defined(KOKKOS_ENABLE_THREADS)
+template <>
+struct FactorizeModeAndAlgo<Kokkos::Threads> : FactorizeModeAndAlgoHostImpl {};
+#endif
+
+#if defined(KOKKOS_ENABLE_OPENMP)
 template <>
-struct FactorizeModeAndAlgo<Kokkos::CudaSpace> {
+struct FactorizeModeAndAlgo<Kokkos::OpenMP> : FactorizeModeAndAlgoHostImpl {};
+#endif
+
+struct FactorizeModeAndAlgoDeviceImpl {
   typedef Mode::Team mode_type;
   typedef Algo::Level3::Unblocked algo_type;
 };
+
+#if defined(KOKKOS_ENABLE_CUDA)
+template <>
+struct FactorizeModeAndAlgo<Kokkos::Cuda> : FactorizeModeAndAlgoDeviceImpl {};
+#endif
+
+#if defined(KOKKOS_ENABLE_HIP)
+template <>
+struct FactorizeModeAndAlgo<Kokkos::Experimental::HIP>
+    : FactorizeModeAndAlgoDeviceImpl {};
 #endif
 
-template <typename ActiveMemorySpace>
+template <typename ExecutionSpace>
 struct SolveModeAndAlgo;
 
-template <>
-struct SolveModeAndAlgo<Kokkos::HostSpace> {
+struct SolveModeAndAlgoHostImpl {
   typedef Mode::Serial mode_type;
   typedef Algo::Level2::Blocked algo_type;
 };
 
-#if defined(KOKKOS_ENABLE_CUDA)
+#if defined(KOKKOS_ENABLE_SERIAL)
 template <>
-struct SolveModeAndAlgo<Kokkos::CudaSpace> {
+struct SolveModeAndAlgo<Kokkos::Serial> : SolveModeAndAlgoHostImpl {};
+#endif
+
+#if defined(KOKKOS_ENABLE_THREADS)
+template <>
+struct SolveModeAndAlgo<Kokkos::Threads> : SolveModeAndAlgoHostImpl {};
+#endif
+
+#if defined(KOKKOS_ENABLE_OPENMP)
+template <>
+struct SolveModeAndAlgo<Kokkos::OpenMP> : SolveModeAndAlgoHostImpl {};
+#endif
+
+struct SolveModeAndAlgoDeviceImpl {
   typedef Mode::Team mode_type;
   typedef Algo::Level2::Unblocked algo_type;
 };
+
+#if defined(KOKKOS_ENABLE_CUDA)
+template <>
+struct SolveModeAndAlgo<Kokkos::Cuda> : SolveModeAndAlgoDeviceImpl {};
+#endif
+
+#if defined(KOKKOS_ENABLE_HIP)
+template <>
+struct SolveModeAndAlgo<Kokkos::Experimental::HIP>
+    : SolveModeAndAlgoDeviceImpl {};
 #endif
 
+template <class VT>
+struct SetTridiagToIdentity {
+ private:
+  VT __AA;
+
+ public:
+  SetTridiagToIdentity(VT AA) : __AA(AA) {}
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const member_type &member) const {
+    const int i = member.league_rank();
+    Kokkos::parallel_for(
+        Kokkos::TeamThreadRange(member, __AA.extent(1)), [&](const int &j) {
+          Kokkos::parallel_for(
+              Kokkos::ThreadVectorRange(member, __AA.extent(5)),
+              [&](const int &v) {
+                for (int k = 0, kend = __AA.extent(3); k < kend; ++k)
+                  __AA(i, j, 1, k, k, v) = 1;
```

Comment:
@e10harvey Is it possible to have a common function to call for this task, like ```TeamSetIdentity<member_type>::invoke()``` in the ```perf_test/batched/KokkosBatched_Test_BlockJacobi_Tutorial.cpp```?

Reply:
Yes. I would add this when needed (if code duplication started).